src/components/ButtonGroup/ButtonGroup.js

```diff
@@ -0,0 +1,93 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import cx from 'classnames';
+import styles from './style.scss';
+
+const defaultIndex = -1;
+const baseClass = 'btn-group';
+
+class ButtonGroup extends React.Component {
+  state = {
+    currentIndex: defaultIndex
+  };
+
+  static getDerivedStateFromProps(props, state) {
+    return {
+      currentIndex:
+        typeof props.currentIndex === 'number'
+          ? props.currentIndex
+          : state.currentIndex
+    };
+  }
+
+  handleClick = index => () => {
+    this.props.onChange(index);
+    this.setState({ currentIndex: index });
+  };
+
+  render() {
+    const { currentIndex } = this.state;
+    const { size, fullWidth, children, className, ...restProps } = this.props;
+
+    const mappedChildren = React.Children.map(children, (child, index) =>
+      React.cloneElement(child, {
+        size,
+        fullWidth,
+        primary: false,
+        destructive: false,
+        onClick: this.handleClick(index),
```

Comment:
It gonna be confusing if some1 pass the `onClick` or any other property which is changed here, but I understand why you did it like that :) We could add a note in the docs

Reply:
well, I've added wrapper lambda which calls `this.handleClick` and then original `child.props.onClick` if provided 🙂